q/protobufkdb.q

```diff
@@ -7,15 +7,39 @@ versionStr:`protobufkdb 2:(`VersionStr;1);
 getMessageSchema:`protobufkdb 2:(`GetMessageSchema;1);
 displayMessageSchema:{[x] -1 getMessageSchema[x];};
 
+getMessageFields:`protobufkdb 2:(`GetMessageFields;1);
+
 addProtoImportPath:`protobufkdb 2:(`AddProtoImportPath;1);
 importProtoFile:`protobufkdb 2:(`ImportProtoFile;1);
 listImportedMessageTypes:`protobufkdb 2:(`ListImportedMessageTypes;1);
 
-saveMessage:`protobufkdb 2:(`SaveMessage;3);
-loadMessage:`protobufkdb 2:(`LoadMessage;2);
-serializeArray:`protobufkdb 2:(`SerializeArray;2);
-parseArray:`protobufkdb 2:(`ParseArray;2);
-serializeArrayArena:`protobufkdb 2:(`SerializeArrayArena;2);
-parseArrayArena:`protobufkdb 2:(`ParseArrayArena;2);
+saveMessage:`protobufkdb 2:(`SaveMessage;4);
+saveMessageFromList:saveMessage[;;;0b];
+saveMessageFromDict:saveMessage[;;;1b];
+
+loadMessage:`protobufkdb 2:(`LoadMessage;3);
+loadMessageToList:loadMessage[;;0b];
+loadMessageToDict:loadMessage[;;1b];
+
+serializeArray:`protobufkdb 2:(`SerializeArray;3);
+serializeArrayFromList:serializeArray[;;0b];
+serializeArrayFromDict:serializeArray[;;1b];
+
+parseArray:`protobufkdb 2:(`ParseArray;3);
+parseArrayToList:parseArray[;;0b];
+parseArrayToDict:parseArray[;;1b];
+
+serializeArrayArena:`protobufkdb 2:(`SerializeArrayArena;3);
+serializeArrayArenaFromList:serializeArrayArena[;;0b];
+serializeArrayArenaFromDict:serializeArrayArena[;;1b];
+
+parseArrayArena:`protobufkdb 2:(`ParseArrayArena;3);
+parseArrayArenaToList:parseArrayArena[;;0b];
+parseArrayArenaToDict:parseArrayArena[;;1b];
+
+LoadMessageDebug:`protobufkdb 2:(`LoadMessageDebug;2);
+loadMessageDebug:{[x;y] -1 LoadMessageDebug[x;y];};
+ParseArrayDebug:`protobufkdb 2:(`ParseArrayDebug;2);
+parseArrayDebug:{[x;y] -1 ParseArrayDebug[x;y];};

```

Comment:
Expose these functions:
```
.protobufkdb.SetStringKdbType :`protobufkdb 2: (`SetStringKdbType;1)
.protobufkdb.GetStringKdbType :`protobufkdb 2: (`GetStringKdbType;1)
.protobufkdb.SetStringMapKeyKdbType :`protobufkdb 2: (`SetStringMapKeyKdbType;1)
.protobufkdb.GetStringMapKeyKdbType :`protobufkdb 2: (`GetStringMapKeyKdbType;1)
```

Reply:
These functions were only used for my testing of -11|4|10h for proto string/bytes fields.  Following feedback from the Ottawa team, we agreed on:

Parsing:

- scalar string = KC
- scalar bytes = KG
- repeated string = 0 of KC
- repeated bytes = 0 of KG
- map-key-string = KS

Serialisation:

These can be interchangeably:
- Scalar string/bytes = -11|4|10h
- Repeated string/bytes = 11 | 0 of KC | 0 of KG

(That's why the code is so ugly now!)

I've now removed these configuration functions from Protobufkdb.h / .cpp